lnx-engine/search-index/src/index.rs

```diff
@@ -150,6 +147,21 @@ impl Index {
     pub async fn destroy(&self) -> Result<()> {
         self.0.destroy().await
     }
+
+    /// Gets the current index document count.
+    pub async fn get_doc_count(&self) -> Result<IndexStats> {
+        let index_meta = self.0._ctx.index.load_metas();
+
+        let mut num_docs: usize = 0;
+        let mut num_deleted_docs: usize = 0;
+
+        for segment in index_meta?.segments {
+            num_docs += segment.num_docs() as usize;
+            num_deleted_docs += segment.num_deleted_docs() as usize;
+        }
+
+        Ok(IndexStats::new(num_docs, num_deleted_docs))
+    }
```

Comment:
This should not be marked as async when it is purely synchronous.

```suggestion
    pub fn get_doc_count(&self) -> Result<IndexStats> {
        let index_meta = self.0._ctx.index.load_metas();

        let mut num_docs: usize = 0;
        let mut num_deleted_docs: usize = 0;

        for segment in index_meta?.segments {
            num_docs += segment.num_docs() as usize;
            num_deleted_docs += segment.num_deleted_docs() as usize;
        }

        Ok(IndexStats::new(num_docs, num_deleted_docs))
    }
```

Reply:
Done. Async removed.